### PR TITLE
chore: update package.json fields and add rimraf

### PR DESCRIPTION
**Pull request by XhmikosR**
**- Summary**

1. added `types` is package.json
2. added rimraf because the build dir wasn't being cleaned and could keep old files

**- Test plan**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,17 +3,17 @@
...
   "files": [
-    "dist/**/*.js",
-    "dist/**/*.d.ts"
+    "dist/**/*.{js,ts}"
   ],
   "scripts": {
     "prepublishOnly": "npm ci && npm test",
     "prepack": "npm run build",
-    "build": "tsc",
+    "build": "rimraf dist && tsc",
```
Comment (XhmikosR) [inline package.json:16]: This change is probably worth having it in all your TS projects to make sure the dist directory doesn't contain any leftovers (probably doesn't apply for automated publishes where the dist dir is clean already).

**Comment by netlify-team-account-1** [comment]
Hi! can you elaborate on the issues this PR fixes? I think typescript types are already working at the moment, and as long as the `.d.ts` files are besides their `.js` files, there's no need to specify the `"types"` property in `package.json`.

File: package.json (hunk 1 above)
Comment (netlify-team-account-1) [reply package.json:16]: We are publishing via CI, so pulling in a new dev dependency is unneccessary in this case.

Comment (XhmikosR) [reply package.json:16]: Cool, I'll revert it I guess :)

**Comment by XhmikosR** [comment]
No problem, types is automatically added by `npm init -y` and I prefer to be explicit personally, but no worries :)

**Pull request closed without merging**